### PR TITLE
Generate broker controller

### PR DESCRIPTION
**Pull request by TheEyePatch**
This PR is mainly for generating the brokers_controller and brokers_controller_spec under request for respec testing. The broker controller is under the admin resources in the routes file so that only an admin can edit, approve and create the Broker model. This PR also includes generating of the basic views of the brokers_controller which can only be accessed by the admin. By generating brokers_controller and its views, it required the admins_controller and its views to be modified to show all the brokers that are approved, and add links to the  specific pages of the brokers views.

### Admin Index (Shows all the approved buyers and brokers)
admin_index

### New Broker template 
broker_new

### Edit Broker template
edit_broker

### List of Pending Brokers 
pending_brokers

### CODE REVIEW

File: app/controllers/admins_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class AdminsController < ApplicationController
   before_action :redirect
   def index
     @admins = Admin.all
-    @brokers = Broker.all
+    @brokers = Broker.where(approved: true)
     @buyers = Buyer.all
```
Comment (TheEyePatch) [inline app/controllers/admins_controller.rb:6]: Will create their own index under admin folder using namespace in the future.

File: app/controllers/brokers_controller.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      redirect_to :edit, @broker.errors.full_messages
+    end
+  end
+
+  private
+
+  def broker_params
+    params.require(:broker).permit(:email, :password, :approved)
+  end
+end
```
Comment (TheEyePatch) [inline app/controllers/brokers_controller.rb:44]: Will add in the admin folder using namespace in the future

File: app/views/admins/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<h1>All Buyers</h1>
+<% @buyers.each do |buyer|%>
+    <div><%= buyer.email%></div>
+<%end%>
+
+<h1>All Brokers</h1>
+<% @brokers.each do |broker|%>
+    <div><%= link_to broker.email, admin_broker_path(current_admin, broker)%><%= link_to 'Edit', edit_admin_broker_path(current_admin, broker)%></div>
+<%end%>
```
Comment (analyn-cajocson) [inline app/views/admins/index.html.erb]: add new line here

File: app/views/brokers/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+  
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<%end%>
```
Comment (analyn-cajocson) [inline app/views/brokers/new.html.erb]: add new line here

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

File: app/views/admins/index.html.erb [hunk 5]
```diff
@@ -7,3 +7,4 @@
 <% @brokers.each do |broker|%>
     <div><%= link_to broker.email, admin_broker_path(current_admin, broker)%><%= link_to 'Edit', edit_admin_broker_path(current_admin, broker)%></div>
 <%end%>
+
```
Comment (TheEyePatch) [inline app/views/admins/index.html.erb:10]: Added newline

File: app/views/brokers/new.html.erb [hunk 6]
```diff
@@ -20,3 +20,4 @@
     <%= f.submit "Create" %>
   </div>
 <%end%>
+
```
Comment (TheEyePatch) [inline app/views/brokers/new.html.erb:23]: Added newline

**Review by analyn-cajocson — APPROVED** [review]

**Pull request merged**